### PR TITLE
Fix reconnect timeout - accept SSH password after network connection reset  #fixed

### LINKED ISSUE #263: Timeout when reconnecting
- Sequel Ace Version: 2.1.3
- macOS Version: 10.15.5 (19F101)
- MySQL Version: 10.1.44-MariaDB-0ubuntu0.18.04.1

**Description**
Occasionally after I've left my Mac alone for a while, and its gone into sleep, or whatever, the SSH MySQL/MariaDB connection is dropped. And when it prompts me for the SSH password (I don't save them for security reasons) I cannot click in the input, or type. I just get the beachball of death. 

The only solution is to kill the Sequel Ace app, and restart it. Annoying when you've been working with large, unsaved queries. It would be handy if they auto saved as you typed.

**Steps To Reproduce**
It is difficult to reproduce. I think it probably has to be a long time asleep. I will test again tomorrow morning.

**Expected Behaviour**
I would expect to be able to enter the requested password

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
Its been in all since I migrated from SequelPro

**Additional Context**
myqlerror

### LINKED ISSUE #658: Connection lost after OS goes to sleep - reconnect doesn't work
V 3.0.1

> Connect to database
> AFK, OS goes to sleep
> Wake up OS and login
> Click on Sequel Ace icon
> Click on any table in current connected DB
> result

Screenshot 2020-12-19 at 10 17 55

my setting

Screenshot 2020-12-19 at 10 18 13

I would except the connection to auto-reconnect or to be alive.

### PR DESCRIPTION
**Pull request by anonymous4argus**
fixed Sequel-Ace issue #263, SequelPro issue #2503

## Changes:
- 

## Closes following issues:
- Closes #263
- Closes #658

## Tested:
- Processors:
  - [X] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [X] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [X] 10.14.x (Mojave)
  - [X] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
- Xcode Version:
- Localizations:
  - [X] English
  - [ ] Spanish
  - [ ] Other (please specify)
## Additional notes:

### CODE REVIEW

**Review by Jason-Morcos — COMMENTED** [review]
Looks good to me, but would like @jamesstout to take a look if possible as he was looking into this issue (if I recall correctly) and he may have some thoughts

File: Source/Other/SSHTunnel/SPSSHTunnel.m [hunk 1]
```diff
@@ -847,6 +847,7 @@ - (void)workerGetPasswordForQuery:(NSString *)theQuery
 	
 	[sshPasswordDialog setFrame:windowFrameRect display:NO];
 	[parentWindow beginSheet:sshPasswordDialog completionHandler:nil];
```
Comment (Kaspik) [inline Source/Other/SSHTunnel/SPSSHTunnel.m:849]: Shouldn't this actually run the modal itself?

Comment (anonymous4argus) [reply Source/Other/SSHTunnel/SPSSHTunnel.m:849]: The modal is not able to accept enough UI interactions when it is invoked with NSModalPanelRunLoopMode during ssh reconnection.

**Review by Kaspik — APPROVED** [review]

**Comment by Kaspik** [comment]
@jamesstout Ready to go?

**Review by jamesstout — APPROVED** [review]

**Pull request merged**